### PR TITLE
Add Site Card edit and delete animations

### LINKED ISSUE #136: Site card edit mode + animation
## Description
Restyle the site card to reflect the latest designs for edit mode. The transition between the "normal" site card and the site card in edit mode should be smooth (the graph should disappear and the buttons should slide in). Also should add the mini graph onto the card. 

Screen Shot 2023-03-21 at 8 43 57 PM

## Todos

These points are rough guidelines. **Please** feel free to discuss the best way to design this component with others on the team!

- [ ] add mini graph onto card
- [ ] restyle site card in edit mode 
- [ ] add transitioning

## Useful links

- [figma](https://www.figma.com/file/Y2IIxfo496bwxFsz8zo9hI/Wireframes?node-id=1096%3A5308&t=A5DbWBDaJDLN2D3U-1)

## Time estimate
tbd!

### PR DESCRIPTION
**Pull request by anyaparekh**
## Status: 
:rocket: Ready

## Description
* Add animation to site card edit mode 
* Disable the edit / done button during transitions to allow the animation to complete

Fixes #136 

## Todos
* Smoothen animations more (??)

## Screenshots
image

### CODE REVIEW

File: components/SideBar.tsx [hunk 1]
```diff
@@ -101,6 +101,18 @@ const Sidebar = () => {
...
+              label="Add a Location"
+              svg={<LocationIcon />}
+              currentPath={router.asPath}
+            />
+            <MenuLink
+              linkProps={{ href: '/form/details' }}
+              label="Edit Site Details"
+              svg={<EditIcon color="#E4E4E4" />}
+              currentPath={router.asPath}
+            />
```
Comment (ericcccsliu) [inline components/SideBar.tsx]: Did we add these back to the sidebar? Think we don't want them there right?

File: components/SiteCard.tsx [hunk 2]
```diff
@@ -1,32 +1,51 @@
...
+      animationElement.current?.addEventListener('transitionend', () => {
+        if (isEditMode) {
+          setDisplayGraph(false);
+        } else {
+          setDisplayGraph(true);
+        }
+      });
+    }, [isEditMode]);
+
+    console.log(displayGraph);
```
Comment (ericcccsliu) [inline components/SiteCard.tsx]: remove this console.log–@priyankajakka

File: pages/sites.tsx [hunk 3]
```diff
@@ -37,11 +38,20 @@ const Sites = () => {
...
       <div className="flex flex-row w-full h-12 items-end mb-4 max-w-lg ">
         <h1 className="flex-1 font-bold text-3xl text-ocf-gray">My Sites</h1>
-        <button onClick={() => setEditMode(!editMode)}>
+        <button
+          onClick={() => {
+            setEditMode(!editMode);
+            setIsDisabled(true);
+            setTimeout(() => {
+              setIsDisabled(false);
+            }, 900);
```
Comment (ericcccsliu) [inline pages/sites.tsx]: lets set this as a constant up above this component, and export it from the file! that way, we can access it wherever else we'll need the transition duration! 🔥

File: components/SiteCard.tsx [hunk 4]
```diff
@@ -53,22 +72,59 @@ const SiteCard = React.forwardRef<HTMLAnchorElement, SiteCardProps>(
...
+          } duration-[900ms]`}
+          ref={animationElement}
+        >
+          {displayGraph && <SiteGraph siteUUID={siteUUID} />}
+        </div>
+
+        <div
+          className={`transition-all ${
+            isEditMode ? 'w-4/12' : 'w-0'
+          } duration-[900ms] flex translate-x-40`}
```
Comment (ericcccsliu) [inline components/SiteCard.tsx]: reference transition duration from `sites` page

File: components/SiteCard.tsx [hunk 5]
```diff
@@ -53,22 +72,59 @@ const SiteCard = React.forwardRef<HTMLAnchorElement, SiteCardProps>(
...
+        >
+          <Link href={'/form/details'} className={`fixed right-0`}>
+            <a
+              className={`w-full flex bg-amber flex-end justify-center ease-in-out transition duration-[900ms] ${
```
Comment (ericcccsliu) [inline components/SiteCard.tsx]: same as above

File: components/SiteCard.tsx [hunk 6]
```diff
@@ -53,22 +72,59 @@ const SiteCard = React.forwardRef<HTMLAnchorElement, SiteCardProps>(
...
+                </div>
+                <p className="flex-1 text-[8px]	text-center px-5">
+                  Edit site details
+                </p>
+              </div>
+            </a>
+          </Link>
+          <Link href={'/form/details'} className={`fixed right-0`}>
+            <a
+              className={`w-full flex bg-[#D44545] flex-end justify-center ease-in-out transition duration-[900ms] ${
```
Comment (ericcccsliu) [inline components/SiteCard.tsx]: same as above

File: components/SiteCard.tsx [hunk 7]
```diff
@@ -53,22 +72,59 @@ const SiteCard = React.forwardRef<HTMLAnchorElement, SiteCardProps>(
...
-            <button>
-              <DeleteIcon />
-            </button>
-          </div>
-        )}
+
+        <div
+          className={`justify-center self-center mr-5 transition-all overflow-hidden max-w-[250px] ${
+            !isEditMode ? 'flex-1' : 'flex-0 w-0'
+          } duration-[900ms]`}
```
Comment (ericcccsliu) [inline components/SiteCard.tsx]: reference transition duration from `sites` page

**Review by ericcccsliu — CHANGES_REQUESTED** [review]

File: components/SiteCard.tsx [hunk 8]
```diff
@@ -1,32 +1,51 @@
...
+      });
+    }, [isEditMode]);
+
+    console.log(displayGraph);
+
     return (
       <a
         onClick={onClick}
         ref={ref}
         href={href}
```
Comment (ericcccsliu) [inline components/SiteCard.tsx]: can we disable this `href` in edit mode?

File: components/SiteCard.tsx (hunk 6 above)
Comment (ericcccsliu) [inline components/SiteCard.tsx]: also, conditionally set `tabindex={0}` here

File: components/SiteCard.tsx (hunk 5 above)
Comment (ericcccsliu) [reply components/SiteCard.tsx]: also set tabindex={0} here conditionally

**Review by ericcccsliu — APPROVED** [review]
🔥 🔥 🔥

File: package.json [hunk 9]
```diff
@@ -25,7 +25,8 @@
     "react-responsive-carousel": "^3.2.23",
     "recharts": "^2.4.3",
     "suncalc": "^1.9.0",
-    "swr": "^2.0.3"
+    "swr": "^2.0.3",
+    "tailwindcss": "^3.2.7"
```
Comment (AndrewLester) [inline package.json]: This is there a reason `tailwindcss` was moved from the `devDependencies` section to the `dependencies` section here?

To move back:
`yarn uninstall tailwindcss`
`yarn add -D tailwindcss`

File: components/icons/EditIcon.tsx [hunk 10]
```diff
@@ -1,9 +1,13 @@
-const EditIcon: React.FC = ({}) => (
+interface props {
+  color: string;
+}
+
+const EditIcon: React.FC<props> = ({ color }) => (
```
Comment (AndrewLester) [inline components/icons/EditIcon.tsx]: Prefer title case for interface names.
```suggestion
interface Props {
  color: string;
}

const EditIcon: React.FC<Props> = ({ color }) => (
```

**Review by AndrewLester — APPROVED** [review]
Still getting that weird negative <rect> value issue, but we can grab that later. Do you want to create an issue for that @anyaparekh?

**Comment by ericcccsliu** [comment]
One last thing—since you haven't merged this in yet, could you add a `.toFixed(2)` on the current output and current capacity numbers? Waaaay too many digits are displaying right now, and this really isn't part of the scope of your issue (and you didn't touch this code in any way) but it's a very trivial change that would really help us out. 
Screen Shot 2023-03-29 at 1 19 25 PM

**Pull request merged**